storage/storage.go

```diff
@@ -19,6 +20,8 @@ var (
 
 	// ErrNoObjectFound indicates there are no objects found from a given directory.
 	ErrNoObjectFound = fmt.Errorf("no object found")
+
+	optionSingle Options
```

Comment:
Can you please try not to keep any global S3 option? Currently, we have both global and non-global options mixed. I think it is quite confusing and difficult to maintain.

Reply:
since `dry-run` is global, i think our option is either to merge `--dry-run` with S3 struct or keep a global option. For `cross-region transfer support` if we will have to keep global options, merging `--dry-run` with those might be better than carrying it with each client. do you have suggestions?

Reply:
In [cross-region PR ](https://github.com/peak/s5cmd/pull/194), you proved that commands can access the flag of parent contexts.

> For cross-region transfer support if we will have to keep global options, merging --dry-run with those might be better than carrying it with each client

I honestly think the opposite. Why do you think this is bad:

```
client := storage.NewClient(src, s3opts)
```

We may discuss it with other contributors to decide which approach is better :D

Reply:
thanks for your suggestion and help. we discussed it with others and decided to do as you suggested - pass --dry-run flag value as an argument rather than keeping a global variable.